SQS connector issue > Can't redeploy my application

Hi there,

Every time I'm trying to redeploy my application, I have the following error message : 

`Caused by: java.lang.IllegalStateException: Timer already cancelled.`

The only way to redeploy my application is to first stop the connector, un-deploy my application, restart the connector and then I can deploy again my application.

Please find enclosed a full stack trace of the logs : [server.log.zip](https://github.com/payara/Cloud-Connectors/files/1250546/server.log.zip)

I tried with the latest Payara (v.173) and amazon-sqs-rar-0.1.0.rar

Thanks for your help,
Regards,
Jerome.

--- a/AmazonSQS/AmazonSQSJCAAPI/src/main/java/fish/payara/cloud/connectors/amazonsqs/api/inbound/SQSPoller.java
+++ b/AmazonSQS/AmazonSQSJCAAPI/src/main/java/fish/payara/cloud/connectors/amazonsqs/api/inbound/SQSPoller.java
@@ -74,28 +74,33 @@ class SQSPoller extends TimerTask {
 
     @Override
     public void run() {
-        ReceiveMessageRequest rmr = new ReceiveMessageRequest(spec.getQueueURL());
-        rmr.setMaxNumberOfMessages(spec.getMaxMessages());
-        rmr.setVisibilityTimeout(spec.getVisibilityTimeout());
-        rmr.setWaitTimeSeconds(spec.getPollInterval()-1);
-        rmr.setAttributeNames(Arrays.asList(spec.getAttributeNames().split(",")));
-        rmr.setMessageAttributeNames(Arrays.asList(spec.getMessageAttributeNames().split(",")));
-        ReceiveMessageResult rmResult = client.receiveMessage(rmr);
-        if (!rmResult.getMessages().isEmpty()) {
-            
-            Class<?> mdbClass = factory.getEndpointClass();
-            for (Message message : rmResult.getMessages()) {
-                for (Method m : mdbClass.getMethods()) {
-                    if (m.isAnnotationPresent(OnSQSMessage.class) && m.getParameterCount() == 1 && m.getParameterTypes()[0].equals(Message.class)) {
-                        try {
-                            ctx.getWorkManager().scheduleWork(new SQSWork(client, factory, m, message,spec.getQueueURL()));
-                        } catch (WorkException ex) {
-                            Logger.getLogger(AmazonSQSResourceAdapter.class.getName()).log(Level.SEVERE, null, ex);
+        try {
+            ReceiveMessageRequest rmr = new ReceiveMessageRequest(spec.getQueueURL());
+            rmr.setMaxNumberOfMessages(spec.getMaxMessages());
+            rmr.setVisibilityTimeout(spec.getVisibilityTimeout());
+            rmr.setWaitTimeSeconds(spec.getPollInterval()-1);
+            rmr.setAttributeNames(Arrays.asList(spec.getAttributeNames().split(",")));
+            rmr.setMessageAttributeNames(Arrays.asList(spec.getMessageAttributeNames().split(",")));
+            ReceiveMessageResult rmResult = client.receiveMessage(rmr);
+            if (!rmResult.getMessages().isEmpty()) {
+
+                Class<?> mdbClass = factory.getEndpointClass();
+                for (Message message : rmResult.getMessages()) {
+                    for (Method m : mdbClass.getMethods()) {
+                        if (m.isAnnotationPresent(OnSQSMessage.class) && m.getParameterCount() == 1 && m.getParameterTypes()[0].equals(Message.class)) {
+                            try {
+                                ctx.getWorkManager().scheduleWork(new SQSWork(client, factory, m, message,spec.getQueueURL()));
+                            } catch (WorkException ex) {
+                                Logger.getLogger(AmazonSQSResourceAdapter.class.getName()).log(Level.SEVERE, null, ex);
+                            }
                         }
                     }
+
                 }
-                
             }
+        } catch (IllegalStateException ise) {
+            // Fix #29 ensure Illegal State Exception doesn't blow up the timer
+            Logger.getLogger(AmazonSQSResourceAdapter.class.getName()).log(Level.WARNING, "Poller caught an Illegal State Exception", ise);
         }
     }
 

